tools/patchwork_integration_test.sh

```diff
@@ -0,0 +1,102 @@
+#! /bin/bash
+set -e
+HOME="$( cd "$( dirname "${BASH_SOURCE[0]}" )" >/dev/null 2>&1 && pwd )"
+PUBIN_PATH=$1
+SINCE=$2
+MBOX=test-mbox.mbox
+PROJECT=test-patchwork
+INITIAL_IMPORT_SIZE=50 # Percentage of mbox to be imported as a raw mbox
+INITIAL_IMPORT_MBOX=initial.mbox
+
+clean_up() {
+	1>/dev/null 2>&1 docker stop pasta-patchwork
+}
+
+trap clean_up EXIT
+
+parse_pubin() {
+	git -C $PUBIN_PATH --no-pager log --pretty=format:%H --since=$SINCE --reverse | \
+		xargs -n 1 -I {} bash -c "
+				git -C $PUBIN_PATH --no-pager show --no-patch --pretty=format:'From %H %ad' {};
+				echo;
+				git -C $PUBIN_PATH --no-pager show {}:m;
+				echo -en '\n\n';
+			"
+}
+
+get_config() {
+	if [ $1 = "mbox" ]
+	then
+		initial_archive=$INITIAL_IMPORT_MBOX
+	else
+		initial_archive=""
+	fi
+	config=$(cat <<-END
+		[PaStA]
+		MODE = "mbox"
+		UPSTREAM = "HEAD^..HEAD"
+
+		[mbox]
+		[mbox.patchwork]
+		url='http://$container_addr:8000/api/'
+		projects = [{id=$project_id, initial_archive="$initial_archive", list_email="test@test.com"}]
+		END
+	)
+}
+
+cd "$HOME"
+
+echo "Parsing pubin..."
+parse_pubin > $MBOX
+if [ ! -d patchwork ]
+then
+	git clone https://github.com/getpatchwork/patchwork.git
+fi
+cp ./pasta-setup.sh patchwork/
+cp ./create_project.py patchwork/
+cp ./$MBOX patchwork/
+cd patchwork
+rm -f *.tar
+echo "UID=$UID" > .env
+echo "GID=`id -g`" >> .env
+docker-compose build
+docker-compose run --rm web ./pasta-setup.sh $MBOX $PROJECT
+
+n=$(( $(wc -l $PROJECT.mbox | cut -d" " -f 1) * $INITIAL_IMPORT_SIZE / 100 ))
+head -n $n $PROJECT.mbox > $INITIAL_IMPORT_MBOX
+mkdir -p ../../resources/linux/resources/mbox/patchwork/
+mv $INITIAL_IMPORT_MBOX $_
+
+echo Starting Patchwork web server...
+docker-compose run --rm --name pasta-patchwork web ./manage.py runserver 0.0.0.0:8000 &
+sleep 5 # Wait for container to start
+container_addr=$(docker inspect -f '{{range .NetworkSettings.Networks}}{{.IPAddress}}{{end}}' pasta-patchwork)
```

Comment:
Yikes... Which means that PaStA's config file depends on the exact IP-address of the container. I think it should be possible to expose the whole patchwork instance to the host with docker-compose ports: https://docs.docker.com/compose/compose-file/#ports

If we do so, we would ensure that we can access 127.0.0.1:8000 from the host, and wouldn't have to pin the IP address of the container.

Reply:
How can we be sure of `127.0.0.1`. What if the script is not run from inside the container? In hydra the container addresses turn out to be of the form 192.168.6.x

Reply:
Please explain your setup: Where do you run PaStA (the host?) and where do you run patchwork (the container?). If so, then pasta could directly access the API via localhost:8000

Reply:
In the commit message, you wrote "ideally it should be run in the container". What if not? Will it fail?

If PaStA runs inside the container, would it be possible to somehow forward the host port where we can access the API? In this case, we could also just access :8000 inside the container.

Reply:
The reason for me saying "ideally it should be run in the container" is that in that case there is no existing index files or mboxes that can mess with the test results. For eg, if the public inbox you are trying to ingest through the script is already imported in pasta the script wont really do much

Reply:
> Please explain your setup: Where do you run PaStA (the host?) and where do you run patchwork (the container?). If so, then pasta could directly access the API via localhost:8000

You can run this script anywhere, doesnt matter whether it's in the container or not. Hence I get the ip address of the container. I dont see a problem with this. What is your concern?

Reply:
I'd prefer if the config file would be independant of the IP address. So I think if there's some other methodology to announce the service to the PaStA Container / Host. Ideally, the config just holds localhost:8000. In case PaStA runs on the host, then it'll work out of the box. If PaStA runs inside the container, then we won't see patchwork via localhost:8000. Could you try if that would be possible with the docker option --net=host?

Reply:
> I'd prefer if the config file would be independant of the IP address

That's possible.

>  If PaStA runs inside the container, then we won't see patchwork via localhost:8000

Not sure about this. Maybe it can work in this case too. So want to test it first.
Was trying to build the pasta docker images but there is some issue, are you able to build the images?

Reply:
The last commit on docker/ was on Aug 5. Back then it worked. What's the issue?

Reply:
`docker-skeleton` doesnt seem to be building. Issues with the ubuntu repository urls I think. Not sure

Reply:
Just did a test build - I can't see an issue.

Reply:
That's weird. I am on Hydra
The command I am using is `docker build -f docker/pasta-skeleton.dockerfile .`